sybil/gate/src/lib.rs

```diff
@@ -43,77 +51,146 @@ pub trait Config: frame_system::Config {
+            request: Vec<Vec<u8>>,
+            requested_response: RequestedSybilResponse
         ) {
             debug::RuntimeLogger::init();
             let sender = ensure_signed(origin)?;
             let location = Junction::Parachain { id: parachain_id };
 
-            // todo: use get the runtime configuration's specific pallet index. Currently, this corresponds
-            // to the index given in our encointer parachain because we declare the module like this in
-            // construct runtime:
-            // `EncointerSybilGate: encointer_sybil_gate::{Module, Call, Storage, Event<T>} = 2,`
-            let sender_pallet_sybil_gate_index = 15u8;
-            // Todo: use actual call_index from proof issuer
-            let call: IssueProofOfPersonhoodConfidenceCall<T::Signature, T::AccountId> = ([pallet_sybil_proof_issuer_index, 0], parachain_id, sender_pallet_sybil_gate_index, request);
+            let request: Vec<ProofOfAttendance<T::Signature, T::AccountId>> =
+                request.into_iter().map(|proof| Decode::decode(&mut proof.as_slice()).unwrap()).collect();
+
+            // Get this pallet's runtime configuration specific module index.
+            let sender_pallet_sybil_gate_index = <T as frame_system::Config>::PalletInfo::index::<Self>()
+                .map(|i| i.checked_into::<u8>())
+                .flatten()
+                .ok_or("[EncointerSybilGate]: PalletIndex does not fix into u8. Consider giving it a smaller index.")?;
+
+            let call = IssueProofOfPersonhoodConfidenceCall::new(
+                pallet_sybil_proof_issuer_index,
+                sender.clone(),
+                request,
+                requested_response,
+                sender_pallet_sybil_gate_index
+            );
+
             let message = Xcm::Transact { origin_type: OriginKind::SovereignAccount, call: call.encode() };
-            debug::debug!(target: LOG, "sending ProofOfPersonhoodRequest to chain: {:?}", parachain_id);
+            debug::debug!(target: LOG, "[EncointerSybilGate]: Sending ProofOfPersonhoodRequest to chain: {:?}", parachain_id);
             match T::XcmSender::send_xcm(location.into(), message) {
-                Ok(()) => Self::deposit_event(RawEvent::ProofOfPersonHoodRequestSentSuccess(sender)),
+                Ok(()) => {
+                    <PendingRequests<T>>::insert(&sender, ());
+                    Self::deposit_event(RawEvent::ProofOfPersonHoodRequestSentSuccess(sender, parachain_id))
+                },
                 Err(e) => Self::deposit_event(RawEvent::ProofOfPersonHoodRequestSentFailure(sender, e)),
             }
         }
 
         #[weight = 5_000_000]
-        fn set_proof_of_personhood_confidence(
+        /// ### Faucet
+        ///
+        /// Faucet that funds accounts. Currently, this can only be called from other parachains, as
+        /// the ProofOfPersonhood can otherwise not be verified.
+        fn faucet(
```

Comment:
a faucet does not belong into this pallet. it should have its own pallet `sybil-gate-template`

Reply:
actually, we could leave it, but rename this entire pallet to `sybil-gate-template` to make sure this code is just an example.

Reply:
Yes, I think it is better to leave it there, but it should be more obvious that this is an example.

Reply:
agreed, see naming comment below